--- a/arch/arm/boot/dts/sun5i-r8-chip.dts	2016-08-31 21:45:47.836244092 +0200
+++ b/arch/arm/boot/dts/sun5i-r8-chip.dts	2016-09-01 00:46:42.831681150 +0200
@@ -56,6 +56,7 @@
 		i2c0 = &i2c0;
 		i2c1 = &i2c1;
 		i2c2 = &i2c2;
+		spi2 = &spi2;
 		serial0 = &uart1;
 		serial1 = &uart3;
 	};
@@ -277,6 +278,19 @@
 	};
 };
 
+&spi2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&spi2_pins_a>;
+	status = "okay";
+	spi2_0 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		compatible = "spidev";
+		reg = <0>;
+		spi-max-frequency = <12000000>;
+	};
+};
+
 &mmc0_pins_a {
 	allwinner,pull = <SUN4I_PINCTRL_PULL_UP>;
 };
