--- a/arch/arm/boot/dts/sun5i-r8-chip.dts	2017-12-02 17:22:31.000000000 +0100
+++ b/arch/arm/boot/dts/sun5i-r8-chip.dts	2017-12-03 15:26:39.410855878 +0100
@@ -58,6 +58,7 @@
 		i2c0 = &i2c0;
 		i2c1 = &i2c1;
 		i2c2 = &i2c2;
+		spi2 = &spi2;
 		serial0 = &uart1;
 		serial1 = &uart3;
 	};
@@ -168,6 +169,19 @@
 	};
 };
 
+&spi2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&spi2_pins_a>, <&spi2_cs0_pins_a>;
+	status = "okay";
+	spi2_0 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		compatible = "spidev";
+		reg = <0>;
+		spi-max-frequency = <12000000>;
+	};
+};
+
 &lradc {
 	vref-supply = <&reg_vcc3v0>;
 	status = "okay";
